app/application/services/search_attractions.rb

```diff
@@ -10,30 +10,45 @@ class SearchAttractions
 
       step :validate_search
       step :retrieve_attractions
+      step :store_attractions
 
       private
 
       API_ERR = 'Cannot access Google Maps API'
+      DB_ERR_MSG = 'Having trouble accessing the database'
 
       def validate_search(input)
-        search = input.call
-        if search.success?
-          Success(search.value!)
+        search_req = input[:search_req].call
+        if search_req.success?
+          Success(input.merge(search_term: search_req.value!))
         else
-          Failure(search.failure)
+          Failure(search_req.failure)
         end
       end
 
       def retrieve_attractions(input)
-        Mapper::AttractionMapper.new(App.config.GMAP_TOKEN).find(input)
-          .then { |attractions| Response::AttractionsList.new(attractions) }
-          .then { |list| Response::ApiResult.new(status: :ok, message: list) }
-          .then { |result| Success(result) }
+        search_term = input[:search_term]
+        attractions = Mapper::AttractionMapper.new(App.config.GMAP_TOKEN).find(search_term)
+        input[:attractions] = attractions
+        Success(input)
       rescue StandardError
         Failure(
           Response::ApiResult.new(status: :internal_error, message: API_ERR)
         )
       end
+
+      def store_attractions(input)
+        attractions = input[:attractions]
+        attractions.each do |attraction|
+          Repository::Attractions.find_or_create(attraction)
+        end
+        attraction_list = Response::AttractionsList.new(attractions)
+        Success(Response::ApiResult.new(status: :ok, message: attraction_list))
+      rescue StandardError
+        Failure(
+          Response::ApiResult.new(status: :internal_error, message: DB_ERR_MSG)
+        )
+      end
```

Comment:
Are we going to store every search results from google api in our database? If yes we will need to implement the "key word" thing in our database too to fully utilize this function and lookup the value inside database first before asking google (right now every request go to google no matter if there are results saved inside our database).

PS. our database is small need to careful

Reply:
Yeah that's my thought, this way we dont need to constantly call google api, I plan on implement the query thing next step.

But i havent thought about the problem of our database being too small, perhaps remove oldest search when full?

Reply:
I first pushed this because I think you might want to start doing this weeks progress and im afraid there might be a conflict

Reply:
But if you need searched result in the database to analyze and recommend its ok to do so. just dont want you to have excessive work , so you can be more focus on the recommendation logic.